state/convert.go

```diff
@@ -568,8 +569,8 @@ func verifyBlockHash(spork *config.Spork, hash []byte, height uint64, hdr *entit
 	}
 	resultHash := flow.MerkleRoot(resultIDs...)
 	resultHashV5 := flow.MerkleRoot(resultIDsV5...)
-	payloadHash := flow.ConcatSum(collectionHash, sealHash, receiptHash, resultHash)
-	payloadHashV5 := flow.ConcatSum(collectionHash, sealHash, receiptHash, resultHashV5)
+	payloadHash := flow.ConcatSum(collectionHash, sealHash, receiptHash, resultHash, toFlowIdentifier(block.ProtocolStateId))
+	payloadHashV5 := flow.ConcatSum(collectionHash, sealHash, receiptHash, resultHashV5, toFlowIdentifier(block.ProtocolStateId))
```

Comment:
I _think_ the way this should work is `payloadHashVN` is the hash of old format, and `payloadHash` is the hash of the latest format. so I think we'd need to differentiate between v5, v6 (the previous version), and the newest version v7.

I think you could do some light refactoring here to make this cleaner. Here's an example of how it's handled for events:
https://github.com/onflow/rosetta/blob/d5760ae4ff8b2b686b9273ecc84d806535d0ada4/state/convert.go#L307-L317

Reply:
[df294a2](https://github.com/onflow/rosetta/pull/59/commits/df294a2a4122ae3c11b5de2a3c49723dfaa01d87)